hasura/graphql-engine
# searchable select ux bug fix (close#5246)

### Description
This PR will fix the behavior of Searchable select component, this will ensure that the unwanted events are ignored and hence SearchInput, Select Dropdown, selected value state  


### Changelog


### Affected components

### Related Issues
close #5246

### Solution and Design
`onMenuClose`, `onBlur`, and `onChange` were triggered almost at the same time, and hence the order of events was not working as expected. (corner cases mentioned in the issue).

This PR will optimize the order by using ref and only emits an event when the component goes to `onBlur` state.



### Steps to test and verify
* Previously selecting an item from the list for the first time was messy, do a comparison.
* Try selecting items multiple times and make sure it's all in sync.
* Leave different user input values and leave the component by clicking outside.

### Limitations, known bugs & workarounds
Maybe a bit of confusion here 
* When there is a selected value, and when the user clears the field, `react-select` shows the last selected value as the placeholder until the component goes to the blurred state.
